### PR TITLE
CI: add a kludge for detecting test failures which try to disguise as successes

### PR DESCRIPTION
**Pull request by ev-br**
Save the output from runtests.py and look for the magic line nose prints for a successful test run. See, e.g., https://github.com/scipy/scipy/issues/4736 for the motivation and idea.

This is not supposed to and does not fix any of the underlying issues discussed in gh-4736. The objective of this PR is just avoid a false positive report from TravisCI.
This PR is successful is the CI run fails with relaxed strides and succeeds for other test runs.

### CODE REVIEW

File: validate_runtests_log.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        expected = expected_size[testmode]
+        actual = int(m.group('num_tests'))
+        if actual < expected:
+            print("Too few tests: expected %s, run %s" % (expected, actual))
+            exit(1)
+        else:
+            exit(0)
+    else:
+        print('Test runner validation errored: did the run really finish?')
+        exit(-1)
```
Comment (argriffing) [inline validate_runtests_log.py]: This is a nitpick, but I think `sys.exit` is preferred in scripts.

**Comment by argriffing** [comment]
This is failing, but in the wrong way.

**Comment by ev-br** [comment]
Have restarted it. Might be that `tee` buffers the output and that sets off Travisbot alarms:-(.

File: validate_runtests_log.py (hunk 1 above)
Comment (pv) [inline validate_runtests_log.py]: The script should also live under `tools/`

**Comment by ev-br** [comment]
OK, with 8cadbad it seems to do the right thing on both py 2 and py 3, now on to simplifying it a bit, then address review comments. Is still WIP.

**Comment by ev-br** [comment]
Okay, it's no longer WIP,  it now it fails in the right way. It's not even too clunky, it seems.  Review comments are addressed as well.

**Comment by pv** [comment]
LGTM, merged

**Comment by rgommers** [comment]
Thanks @ev-br!

**Pull request merged**